R/aggregate_game_stats_kicking.R

```diff
@@ -0,0 +1,185 @@
+#' Summarize Kicking Stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#' @param weekly If `TRUE`, returns week-by-week stats, otherwise, stats for
+#' the entire data frame in argument `pbp`.
+#'
+#' @examples
+#' \donttest{
+#' try({
+#'     pbp <- nflreadr::load_pbp(2021)
+#'     weekly <- calculate_player_stats_kicking(pbp, weekly = TRUE)
+#'     dplyr::glimpse(weekly)
+#'
+#'     overall <- calculate_player_stats_kicking(pbp, weekly = FALSE)
+#'     dplyr::glimpse(overall)
+#' })
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_player_stats_kicking <- function(pbp, weekly = FALSE) {
+
+  # need newer version of nflreadr to use load_players
+  rlang::check_installed("nflreadr (>= 1.3.0)")
+
+  # First, creating a grouping variable object to toggle the weekly argument w/
+  grp_vars <- if (isTRUE(weekly)){
+    list("season", "week", "season_type", "player_id", "team")
+  } else if (isFALSE(weekly)){
+    list("player_id", "team")
```

Comment:
we do not group by team if weekly = FALSE in the other stats functions. We may or may not change this in the future but I would like to be consistent for now

Reply:
Offense and defense are inconsistent here now. I think I prefer team-separated as per defense PR